File: C++/Product.cpp
/* 
    Saya Alif Faturahman Firdaus (2107377) mengerjakan Latihan Praktikum 2 dalam mata 
    kuliah DPBO untuk keberkahan-Nya maka saya tidak melakukan kecurangan seperti yang 
    telah dispesifikasikan. Aamiin.
*/

// ----- Latihan C++ ----- //

// Import library.
#include <bits/stdc++.h>

// Using standard namespace.
using namespace std;

class Product
{
    // Membuat atribut privte
private:
    // id Product
    string id;
    // name Product
    string name;
    // brand Product
    string brand;
    // jenis kelamin Product
    string price;

public:
    Product() {} // Konstruktor Product

    Product(string id, string brand, string name, string price)
    {
        // Konstruktor langsung isi attribut
        this->id = id;
        this->name = name;
        this->brand = brand;
        this->price = price;
    }

    // Setter and Getter methods for private attributes

    // Setter ID
    void setID(string id)
    {
        this->id = id;
    }

    // Getter ID
    string getID()
    {
        return id;
    }
    
    // Setter Name
    void setName(string name)
    {
        this->name = name;
    }

    // Getter Name
    string getName()
    {
        return name;
    }

    // Setter Brand
    void setBrand(string brandProduct)
    {
        this->brand = brandProduct;
    }

    // Getter Brand
    string getBrand()
    {
        return brand;
    }

    // Setter Price
    void setPrice(string price)
    {
        this->price = price;
    }

    // Getter Price
    string getPrice()
    {
        return this->price;
    }

    // Destruktor
    ~Product()
    {
    }
};

File: Java/Product.java
/* 
    Saya Alif Faturahman Firdaus (2107377) mengerjakan Latihan Praktikum 2 dalam mata 
    kuliah DPBO untuk keberkahan-Nya maka saya tidak melakukan kecurangan seperti yang 
    telah dispesifikasikan. Aamiin.
*/

// ----- Latihan Java ----- //

public class Product {
    // ID Product Product
    private String idProduct;
    // Name Product
    private String name;
    // Brand Product
    private String brand;
    // Price Product
    private String price;


    Product(){} // konstruktor Product

    Product(String idProduct, String name, String brand, String price) {
        // konstruktor langsung isi attribut
        this.idProduct = idProduct;
        this.name = name;
        this.brand = brand;
        this.price = price;
    }
    
    // set idProduct
    public void setIdProduct(String idProduct) {
        this.idProduct = idProduct;
    }

    // get idProduct
    public String getIdProduct() {
        return idProduct;
    }

    // set name
    public void setName(String nameProduct) {
        this.name = nameProduct;
    }

    // get name
    public String getName() {
        return name;
    }

    // set Brand
    public void setBrand(String brand) {
        this.brand = brand;
    }

    // get Brand
    public String getBrand() {
        return this.brand;
    }

    // set Price
    public void setPrice(String price) {
        this.price = price;
    }

    // get Price
    public String getPrice() {
        return this.price;
    }

}